### PR TITLE
Fix/enable client id and secret credentials

### PR DESCRIPTION
**Pull request by DanielPDWalker**
- Added conditional logic around initializing authhttp, as you can only do it using oauth credentials.
- Added conditional logic around creating service build() as again oauth credentials and client_id and client_secret differ.
- Updated conditional logic around selecting which credentials to use.
- Added test "test_client_id_and_client_secret_with_refresh_proxy_url_discover" to make sure the updated logic around selecting credentials work.
- Updated test_core to set up self.mock_http for each test within the test.
- Updated gitignore to ignore .vscode files.

### CODE REVIEW

File: tap_google_analytics/ga_client.py [hunk 1]
```diff
@@ -93,19 +93,25 @@ def __init__(self, config):
...
 
     def initialize_credentials(self, config):
         if config.get('authorization', {}).get('bearer_token', None):
             self.authorization = config['authorization']
             return Credentials(token=config['authorization']['bearer_token'], refresh_handler=self.noop_refresh_handler)
         elif config.get('oauth_credentials', {}).get('refresh_proxy_url', None) \
                 and config.get('oauth_credentials', {}).get('access_token', None) \
-                and config.get('oauth_credentials', {}).get('refresh_token', None):
+                and config.get('oauth_credentials', {}).get('refresh_token', None) \
+                and config.get('oauth_credentials', {}).get('client_id') == None \
```
Comment (aphethean1) [inline tap_google_analytics/ga_client.py:113]: Aren't these identical, why the change of style?

                and config.get('oauth_credentials', {}).get('refresh_token', None) \
                and config.get('oauth_credentials', {}).get('client_id') == None \
                and config.get('oauth credentials', {}).get('client_secret') == None:

File: tap_google_analytics/ga_client.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 from googleapiclient.errors import HttpError
 
 from oauth2client.service_account import ServiceAccountCredentials
-from oauth2client.client import GoogleCredentials
+from oauth2client.client import GoogleCredentials, _raise_exception_for_missing_fields
```
Comment (aphethean1) [inline tap_google_analytics/ga_client.py]: Remove unused import

**Review by aphethean1 — CHANGES_REQUESTED** [review]

File: tap_google_analytics/ga_client.py (hunk 1 above)
Comment (DanielPDWalker) [reply tap_google_analytics/ga_client.py:113]: The logic has to be like this as you will always have these 5 credentials with both the oauth and client_id and client_secret ways. 

By making sure that client_id and client_secret are specifically None, we know that we aren't using the client_id and client_secret credentials. 

If you do it the .get('client_id', None) way then if you are using client_id and client_secret you will always get into that conditional as all 5 are true for that version, when we just want to get there is there is no bearer_token, from the last if statement, and if we are specifically not supplying client_id and client_secret.

File: tap_google_analytics/ga_client.py (hunk 2 above)
Comment (DanielPDWalker) [reply tap_google_analytics/ga_client.py]: Ah missed this, it was when I trying to handle a credentials exception. Speaking of I've changed the sys.exit in the conditionals around the build() to use the LOGGER and then sys.exit(1), the bring it inline with the rest of the code.

**Comment by aphethean1** [comment]
Let's reopen
